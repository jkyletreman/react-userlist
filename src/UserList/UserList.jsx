import React from "react";
import UserData from "../data/context";

export const UserList = () => {
  return (
    <UserData.Consumer>
      {context => {
        return context.state.userData.map(user => {
          return (
            <li key={user.id} onClick={context.selectUser}>
              {user.first} {user.last}
            </li>
          );
        });
      }}
    </UserData.Consumer>
  );
};

export default UserList;
