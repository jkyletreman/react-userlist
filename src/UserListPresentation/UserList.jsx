import React from "react";
import UserData from "../context";

const UserList = ({ userData, selectUser}) => {
  return (
    <ul>
      {userData.map(user => {
        return (
          <li key={user.id} onClick={selectUser}>
            {user.first} {user.last}
          </li>
        );
      })}
    </ul>
  );
};

export default UserList;
