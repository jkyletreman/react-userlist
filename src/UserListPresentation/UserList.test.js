import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import UserList from './UserList'
import ShallowRenderer from 'react-test-renderer/shallow';

it('renders user shallow correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<UserList userData={userData} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});

const userData = [
  {
    id: 1,
    first: "John",
    last: "Roberts",
    age: 51,
    location: "Chicago, IL",
    description: "John is a retired YouTuber"
  },
  {
    id: 2,
    first: "Frita",
    last: "Waters",
    age: 27,
    location: "Phoenix, AZ",
    description: "Frita loves the desert, cheese, and turtles."
  },
  {
    id: 3,
    first: "Bart",
    last: "Simpson",
    age: 24,
    location: "Springfield, IL",
    description: "Bart skateboards everywhere."
  }
];
