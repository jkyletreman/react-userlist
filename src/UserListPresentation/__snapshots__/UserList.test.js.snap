// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user shallow correctly 1`] = `
<ul>
  <li
    onClick={undefined}
  >
    John
     
    Roberts
  </li>
  <li
    onClick={undefined}
  >
    Frita
     
    Waters
  </li>
  <li
    onClick={undefined}
  >
    Bart
     
    Simpson
  </li>
</ul>
`;
